pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -651,10 +679,36 @@ func conditionallyConvertToIstioMtls(
 	serviceAccounts []string,
 	sni string,
 	proxy *model.Proxy,
+	tlsPolicy *authn.Policy,
+	mtlsReady bool,
 ) *networking.TLSSettings {
+	if features.UseAutoPilotMTLS.Get() && tlsPolicy != nil {
```

Comment:
I would prefer the processing of the policy details, peer authentication to be put into authn related package. and return a boolean indicating whether autopilot should be kicked in.

such that cluster.go file don't won't coupled with server side policy.